### ISSUE TITLE
[1.17] Stuck on waiting for chunks after going through nether portal 

### ISSUE THREAD

**Issue by ghost** [PaperMC/Paper#5877]
### Timings or Profile link

https://timings.aikar.co/?id=4027290e45cd4afdb41c54977bf5422c#timings

### Description of issue

Most times I go through a nether portal, both overworld to nether and nether to overworld, the world is blank and the client says "Waiting for chunk" in F3.

### Plugin list

none

### Server config files

```
# This is the main configuration file for Bukkit.
# As you can see, there's actually not that much to configure without any plugins.
# For a reference for any variable inside this file, check out the Bukkit Wiki at
# https://www.spigotmc.org/go/bukkit-yml
# 
# If you need help on this file, feel free to join us on irc or leave a message
# on the forums asking for advice.
# 
# IRC: #spigot @ irc.spi.gt
#    (If this means nothing to you, just go to https://www.spigotmc.org/go/irc )
# Forums: https://www.spigotmc.org/
# Bug tracker: https://www.spigotmc.org/go/bugs


settings:
  allow-end: true
  warn-on-overload: true
  permissions-file: permissions.yml
  update-folder: update
  plugin-profiling: false
  connection-throttle: 4000
  query-plugins: true
  deprecated-verbose: default
  shutdown-message: Server closed
  minimum-api: none
spawn-limits:
  monsters: 70
  animals: 10
  water-animals: 5
  water-ambient: 20
  ambient: 15
chunk-gc:
  period-in-ticks: 600
ticks-per:
  animal-spawns: 400
  monster-spawns: 1
  water-spawns: 1
  water-ambient-spawns: 1
  ambient-spawns: 1
  autosave: 6000
aliases: now-in-commands.yml
```

```
# This is the main configuration file for Spigot.
# As you can see, there's tons to configure. Some options may impact gameplay, so use
# with caution, and make sure you know what each option does before configuring.
# For a reference for any variable inside this file, check out the Spigot wiki at
# http://www.spigotmc.org/wiki/spigot-configuration/
# 
# If you need help with the configuration or have any questions related to Spigot,
# join us at the IRC or drop by our forums and leave a post.
# 
# IRC: #spigot @ irc.spi.gt ( http://www.spigotmc.org/pages/irc/ )
# Forums: http://www.spigotmc.org/

config-version: 12
settings:
  debug: false
  bungeecord: false
  sample-count: 12
  player-shuffle: 0
  user-cache-size: 1000
  save-user-cache-on-stop-only: false
  moved-wrongly-threshold: 0.0625
  moved-too-quickly-multiplier: 10.0
  log-villager-deaths: true
  timeout-time: 60
  restart-on-crash: true
  restart-script: ./start.sh
  netty-threads: 4
  attribute:
    maxHealth:
      max: 2048.0
    movementSpeed:
      max: 2048.0
    attackDamage:
      max: 2048.0
messages:
  whitelist: You are not whitelisted on this server!
  unknown-command: Unknown command. Type "/help" for help.
  server-full: The server is full!
  outdated-client: Outdated client! Please use {0}
  outdated-server: Outdated server! I'm still on {0}
  restart: Server is restarting
advancements:
  disable-saving: false
  disabled:
  - minecraft:story/disabled
commands:
  replace-commands:
  - setblock
  - summon
  - testforblock
  - tellraw
  log: true
  tab-complete: 0
  send-namespaced: true
  spam-exclusions:
  - /skill
  silent-commandblock-console: false
stats:
  disable-saving: false
  forced-stats: {}
players:
  disable-saving: false
world-settings:
  default:
    verbose: false
    hopper-amount: 1
    seed-village: 10387312
    seed-desert: 14357617
    seed-igloo: 14357618
    seed-jungle: 14357619
    seed-swamp: 14357620
    seed-monument: 10387313
    seed-shipwreck: 165745295
    seed-ocean: 14357621
    seed-outpost: 165745296
    seed-endcity: 10387313
    seed-slime: 987234911
    seed-bastion: 30084232
    seed-fortress: 30084232
    seed-mansion: 10387319
    seed-fossil: 14357921
    seed-portal: 34222645
    max-tnt-per-tick: 100
    enable-zombie-pigmen-portal-spawns: true
    item-despawn-rate: 6000
    view-distance: default
    wither-spawn-sound-radius: 0
    arrow-despawn-rate: 1200
    trident-despawn-rate: 1200
    hanging-tick-frequency: 100
    zombie-aggressive-towards-villager: true
    nerf-spawner-mobs: false
    mob-spawn-range: 8
    end-portal-sound-radius: 0
    dragon-death-sound-radius: 0
    max-entity-collisions: 8
    entity-tracking-range:
      players: 48
      animals: 48
      monsters: 48
      misc: 32
      other: 64
    ticks-per:
      hopper-transfer: 8
      hopper-check: 1
    hunger:
      jump-walk-exhaustion: 0.05
      jump-sprint-exhaustion: 0.2
      combat-exhaustion: 0.1
      regen-exhaustion: 6.0
      swim-multiplier: 0.01
      sprint-multiplier: 0.1
      other-multiplier: 0.0
    max-tick-time:
      tile: 50
      entity: 50
    squid-spawn-range:
      min: 45.0
    merge-radius:
      exp: 3.0
      item: 2.5
    growth:
      cactus-modifier: 100
      cane-modifier: 100
      melon-modifier: 100
      mushroom-modifier: 100
      pumpkin-modifier: 100
      sapling-modifier: 100
      beetroot-modifier: 100
      carrot-modifier: 100
      potato-modifier: 100
      wheat-modifier: 100
      netherwart-modifier: 100
      vine-modifier: 100
      cocoa-modifier: 100
      bamboo-modifier: 100
      sweetberry-modifier: 100
      kelp-modifier: 100
    entity-activation-range:
      animals: 32
      monsters: 32
      raiders: 48
      misc: 16
      water: 16
      villagers: 32
      flying-monsters: 32
      villagers-work-immunity-after: 100
      villagers-work-immunity-for: 20
      villagers-active-for-panic: true
      tick-inactive-villagers: true
      wake-up-inactive:
        animals-max-per-tick: 4
        animals-every: 1200
        animals-for: 100
        monsters-max-per-tick: 8
        monsters-every: 400
        monsters-for: 100
        villagers-max-per-tick: 4
        villagers-every: 600
        villagers-for: 100
        flying-monsters-max-per-tick: 8
        flying-monsters-every: 200
        flying-monsters-for: 100
```
```
#Minecraft server properties
#Sat Jun 19 07:19:20 BST 2021
enable-jmx-monitoring=false
rcon.port=25575
gamemode=survival
enable-command-block=false
enable-query=false
level-name=world
motd=A Minecraft Server
query.port=25565
pvp=true
difficulty=easy
network-compression-threshold=256
require-resource-pack=false
max-tick-time=60000
use-native-transport=true
max-players=20
online-mode=true
enable-status=true
allow-flight=false
broadcast-rcon-to-ops=true
view-distance=10
server-ip=
resource-pack-prompt=
allow-nether=true
server-port=25565
enable-rcon=false
sync-chunk-writes=true
op-permission-level=4
prevent-proxy-connections=false
resource-pack=
entity-broadcast-range-percentage=100
rcon.password=
player-idle-timeout=0
debug=false
force-gamemode=false
rate-limit=0
hardcore=false
white-list=false
broadcast-console-to-ops=true
spawn-npcs=true
spawn-animals=true
snooper-enabled=true
function-permission-level=2
text-filtering-config=
spawn-monsters=true
enforce-whitelist=false
resource-pack-sha1=
spawn-protection=16
max-world-size=29999984
```

### Paper version

This server is running Paper version git-Paper-"309b99b" (MC: 1.17) (Implementing API version 1.17-R0.1-SNAPSHOT)
Error obtaining version information

### Agreements

- [X] I am running the latest version of Paper available from https://papermc.io/downloads.
- [X] I have searched for and ensured there isn't already an open issue regarding this.
- [X] My version of Minecraft is supported by Paper.

**Comment by ghost** [comment]
Here are the profiling results using YourKit

https://filebin.net/yldd3lus99gcaypn

```
[09:15:18 INFO]: BikeShedFactory joined the game
[09:15:18 INFO]: BikeShedFactory[/127.0.0.1:34342] logged in with entity id 60 at ([world]-2123.5, 63.174430555484136, -369.0987426304581)
[09:16:22 WARN]: Can't keep up! Is the server overloaded? Running 5016ms or 100 ticks behind
[09:17:35 WARN]: Can't keep up! Is the server overloaded? Running 42565ms or 851 ticks behind
[09:18:47 ERROR]: --- DO NOT REPORT THIS TO PAPER - THIS IS NOT A BUG OR A CRASH  - git-Paper-"309b99b" (MC: 1.17) ---
[09:18:47 ERROR]: The server has not responded for 10 seconds! Creating thread dump
[09:18:47 ERROR]: ------------------------------
[09:18:47 ERROR]: Server thread dump (Look for plugins here before reporting to Paper!):
[09:18:47 ERROR]: ------------------------------
[09:18:47 ERROR]: Current Thread: Server thread
[09:18:47 ERROR]: 	PID: 27 | Suspended: false | Native: true | State: RUNNABLE
[09:18:47 ERROR]: 	Stack:
[09:18:47 ERROR]: 		java.base@16-ea/java.lang.invoke.DirectMethodHandle.allocateInstance(DirectMethodHandle.java:521)
[09:18:47 ERROR]: 		java.base@16-ea/java.lang.invoke.DirectMethodHandle$Holder.newInvokeSpecial(DirectMethodHandle$Holder)
[09:18:47 ERROR]: 		java.base@16-ea/java.lang.invoke.Invokers$Holder.linkToTargetMethod(Invokers$Holder)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ChunkMap.getChunkRangeFuture(ChunkMap.java:795)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ChunkMap.scheduleChunkGeneration(ChunkMap.java:1276)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ChunkMap.schedule(ChunkMap.java:1183)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ChunkHolder.getOrScheduleFuture(ChunkHolder.java:547)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ChunkMap.schedule(ChunkMap.java:1173)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ChunkHolder.getOrScheduleFuture(ChunkHolder.java:547)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ChunkMap.schedule(ChunkMap.java:1173)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ChunkHolder.getOrScheduleFuture(ChunkHolder.java:547)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ChunkMap.schedule(ChunkMap.java:1173)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ChunkHolder.getOrScheduleFuture(ChunkHolder.java:547)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ChunkMap.schedule(ChunkMap.java:1173)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ChunkHolder.getOrScheduleFuture(ChunkHolder.java:547)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ChunkMap.schedule(ChunkMap.java:1173)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ChunkHolder.getOrScheduleFuture(ChunkHolder.java:547)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ChunkMap.getChunkRangeFuture(ChunkMap.java:791)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ChunkMap.scheduleChunkGeneration(ChunkMap.java:1276)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ChunkMap.schedule(ChunkMap.java:1183)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ChunkHolder.getOrScheduleFuture(ChunkHolder.java:547)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ChunkMap.getChunkRangeFuture(ChunkMap.java:791)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ChunkMap.prepareAccessibleChunk(ChunkMap.java:1468)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ChunkHolder.updateFutures(ChunkHolder.java:687)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.DistanceManager.runAllUpdates(DistanceManager.java:130)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ServerChunkCache.runDistanceManagerUpdates(ServerChunkCache.java:667)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ServerChunkCache.saveIncrementally(ServerChunkCache.java:709)
[09:18:47 ERROR]: 		app//net.minecraft.server.level.ServerLevel.saveIncrementally(ServerLevel.java:1021)
[09:18:47 ERROR]: 		app//net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1435)
[09:18:47 ERROR]: 		app//net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1180)
[09:18:47 ERROR]: 		app//net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:320)
[09:18:47 ERROR]: 		app//net.minecraft.server.MinecraftServer$$Lambda$3665/0x000000080087e8f8.run(Unknown Source)
[09:18:47 ERROR]: 		java.base@16-ea/java.lang.Thread.run(Thread.java:831)
[09:18:47 ERROR]: ------------------------------
[09:18:47 ERROR]: --- DO NOT REPORT THIS TO PAPER - THIS IS NOT A BUG OR A CRASH ---
[09:18:47 ERROR]: ------------------------------
[09:18:57 INFO]: BikeShedFactory lost connection: Disconnected
[09:18:57 INFO]: BikeShedFactory left the game
[09:19:07 WARN]: Can't keep up! Is the server overloaded? Running 62060ms or 1241 ticks behind
[09:19:42 ERROR]: --- DO NOT REPORT THIS TO PAPER - THIS IS NOT A BUG OR A CRASH  - git-Paper-"309b99b" (MC: 1.17) ---
[09:19:42 ERROR]: The server has not responded for 24 seconds! Creating thread dump
[09:19:42 ERROR]: ------------------------------
[09:19:42 ERROR]: Server thread dump (Look for plugins here before reporting to Paper!):
[09:19:42 ERROR]: ------------------------------
[09:19:42 ERROR]: Current Thread: Server thread
[09:19:42 ERROR]: 	PID: 27 | Suspended: false | Native: true | State: RUNNABLE
[09:19:42 ERROR]: 	Stack:
[09:19:42 ERROR]: 		app//it.unimi.dsi.fastutil.longs.LongAVLTreeSet.subSet(LongAVLTreeSet.java:1078)
[09:19:42 ERROR]: 		app//net.minecraft.world.level.entity.EntitySectionStorage.forEachAccessibleSection(EntitySectionStorage.java:47)
[09:19:42 ERROR]: 		app//net.minecraft.world.level.entity.EntitySectionStorage.getEntities(EntitySectionStorage.java:126)
[09:19:42 ERROR]: 		app//net.minecraft.world.level.entity.LevelEntityGetterAdapter.get(LevelEntityGetterAdapter.java:46)
[09:19:42 ERROR]: 		app//net.minecraft.world.level.Level.getEntities(Level.java:989)
[09:19:42 ERROR]: 		app//net.minecraft.world.level.EntityGetter.getEntityCollisions(EntityGetter.java:58)
[09:19:42 ERROR]: 		app//net.minecraft.world.level.CommonLevelAccessor.getEntityCollisions(CommonLevelAccessor.java:28)
[09:19:42 ERROR]: 		app//net.minecraft.world.entity.Entity.collide(Entity.java:1081)
[09:19:42 ERROR]: 		app//net.minecraft.world.entity.Entity.move(Entity.java:844)
[09:19:42 ERROR]: 		app//net.minecraft.world.entity.LivingEntity.handleRelativeFrictionAndCalculateMovement(LivingEntity.java:2771)
[09:19:42 ERROR]: 		app//net.minecraft.world.entity.LivingEntity.travel(LivingEntity.java:2726)
[09:19:42 ERROR]: 		app//net.minecraft.world.entity.monster.Strider.travelWithInput(Strider.java:284)
[09:19:42 ERROR]: 		app//net.minecraft.world.entity.ItemSteerable.travel(ItemSteerable.java:25)
[09:19:42 ERROR]: 		app//net.minecraft.world.entity.monster.Strider.travel(Strider.java:270)
[09:19:42 ERROR]: 		app//net.minecraft.world.entity.LivingEntity.aiStep(LivingEntity.java:3188)
[09:19:42 ERROR]: 		app//net.minecraft.world.entity.Mob.aiStep(Mob.java:620)
[09:19:42 ERROR]: 		app//net.minecraft.world.entity.AgeableMob.aiStep(AgeableMob.java:154)
[09:19:42 ERROR]: 		app//net.minecraft.world.entity.animal.Animal.aiStep(Animal.java:59)
[09:19:42 ERROR]: 		app//net.minecraft.world.entity.LivingEntity.tick(LivingEntity.java:2874)
[09:19:42 ERROR]: 		app//net.minecraft.world.entity.Mob.tick(Mob.java:388)
[09:19:42 ERROR]: 		app//net.minecraft.world.entity.monster.Strider.tick(Strider.java:330)
[09:19:42 ERROR]: 		app//net.minecraft.world.entity.Entity.rideTick(Entity.java:2262)
[09:19:42 ERROR]: 		app//net.minecraft.world.entity.LivingEntity.rideTick(LivingEntity.java:3378)
[09:19:42 ERROR]: 		app//net.minecraft.server.level.ServerLevel.tickPassenger(ServerLevel.java:977)
[09:19:42 ERROR]: 		app//net.minecraft.server.level.ServerLevel.tickNonPassenger(ServerLevel.java:952)
[09:19:42 ERROR]: 		app//net.minecraft.server.level.ServerLevel$$Lambda$4465/0x0000000800c930b8.accept(Unknown Source)
[09:19:42 ERROR]: 		app//net.minecraft.world.level.Level.guardEntityTick(Level.java:857)
[09:19:42 ERROR]: 		app//net.minecraft.server.level.ServerLevel.lambda$tick$5(ServerLevel.java:639)
[09:19:42 ERROR]: 		app//net.minecraft.server.level.ServerLevel$$Lambda$4445/0x0000000800d85010.accept(Unknown Source)
[09:19:42 ERROR]: 		app//net.minecraft.world.level.entity.EntityTickList.forEach(EntityTickList.java:54)
[09:19:42 ERROR]: 		app//net.minecraft.server.level.ServerLevel.tick(ServerLevel.java:620)
[09:19:42 ERROR]: 		app//net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1551)
[09:19:42 ERROR]: 		app//net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:476)
[09:19:42 ERROR]: 		app//net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:1404)
[09:19:42 ERROR]: 		app//net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1180)
[09:19:42 ERROR]: 		app//net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:320)
[09:19:42 ERROR]: 		app//net.minecraft.server.MinecraftServer$$Lambda$3665/0x000000080087e8f8.run(Unknown Source)
[09:19:42 ERROR]: 		java.base@16-ea/java.lang.Thread.run(Thread.java:831)
[09:19:42 ERROR]: ------------------------------
[09:19:42 ERROR]: --- DO NOT REPORT THIS TO PAPER - THIS IS NOT A BUG OR A CRASH ---
[09:19:42 ERROR]: ------------------------------
[09:19:45 WARN]: Can't keep up! Is the server overloaded? Running 8653ms or 173 ticks behind
```

**Comment by memikri** [comment]
+1 also experiencing this issue

**Comment by DoubleGe** [comment]
Same I also have this issue

**Comment by JblackMC** [comment]
Confirmed that I am having the same issue

**Comment by Chew** [comment]
If you can reproduce it please provide more info than just "same":
1) Is this a new world or an upgraded world?
2) Any plugins?
3) What is your exact `/version` output?

**Comment by memikri** [comment]
I'm having the same issue on a new world, with no plugins, on Paper build 5

**Comment by kennykantisiri** [comment]
I am also able to reproduce this issue.

- Completely new world
- No Plugins
- Paper Build `#17` (33d6b6f)

**Comment by e-im** [comment]
Please leave a reaction on the initial comment rather than creating a new comment if you are able to reproduce, unless you have new information to add.

**Comment by SG3xHERO** [comment]
I am also having this issue on my survival world that was previously  1.16.5

**Comment by TechnicallyCoded** [comment]
I also had this issue and was redirected to #5910

**Comment by Machine-Maker** [comment]
Is this still happening on >= build 46?

**Comment by Bilibox** [comment]
I am able to reproduce this issue.

Completely new world
No Plugins
Paper Build 46

**Comment by Oxxzyn** [comment]
I've been unable to replicate this on Paper Build 46 as compared to previous versions.

**Comment by ghost** [comment]
I am now unable to replicate this on build 53

**Comment by Bilibox** [comment]
For build 53
The issue has now basically been resolved. 
I only see "waiting for chunks" for a max of about 3 seconds from my small testing which is a huge improvement (and could also just be lag).

**Comment by Crousus** [comment]
I'm also on build 53 and i have a similar issue i think. Basically everytime somebody teleports into a unloaded area or joins the server, the server stands still for ~2secs. It also takes this long for any Chunks to load. In the Timings Viewer this appears as "Sync Chunk Load" i believe.

https://timings.aikar.co/?id=eba2aae4be4345cf8a4c1b4782072eea

**Comment by TechnicallyCoded** [comment]
Same deal here, mostly after going through nether portals. The chunks around the one I am standing in load first. Then, eventually, the chunk I am standing is sent to the client and loads. Once that is done the chunks seem to load perfectly fine and fast, just not after a teleport for whatever reason.

Edit: Tested with 1.17 build 64

**Comment by e-im** [comment]
Generally no longer relevant especially as Tuinity has been merged into Paper since 1.17.1. Please open a new issue or leave a comment should you still be experiencing this on 1.18.2.
